spec/hashie/mash_spec.rb

```diff
@@ -174,6 +173,50 @@
 
       expect(logger_output).to be_blank
     end
+
+    it 'writes to logger when a key is overridden that is not blacklisted' do
+      mash_class = Class.new(Hashie::Mash) do
+        disable_warnings :merge
+      end
+
+      mash_class.new('address' => { 'zip' => '90210' })
+      expect(logger_output).not_to be_blank
+    end
+
+    it 'does not write to logger when a key is overridden that is blacklisted' do
+      mash_class = Class.new(Hashie::Mash) do
+        disable_warnings :zip
+      end
+
+      mash_class.new('address' => { 'zip' => '90210' })
+      expect(logger_output).to be_blank
+    end
+
+    it 'carries over the disabled blacklist for warnings on grandchild classes' do
+      child_class = Class.new(Hashie::Mash) do
+        disable_warnings :zip, :merge
+      end
+      grandchild_class = Class.new(child_class)
+
+      grandchild_class.new('address' => { 'zip' => '90210' })
+
+      expect(grandchild_class.disable_warnings_blacklist.count).to eq(2)
+      expect(logger_output).to be_blank
+    end
+
+    it 'allows grandchild classes to be more strict than child' do
+      child_class = Class.new(Hashie::Mash) do
+        disable_warnings :zip
+      end
+
+      grandchild_class = Class.new(child_class) do
+        disable_warnings
+      end
+
+      grandchild_class.new('merge' => 'true')
+
+      expect(logger_output).to be_blank
+    end
```

Comment:
I think we need to cover more use-case combinations, especially with multiple calls to `disable_warnings`

Reply:
359ac81d029c5ae09e11e7b1aa75beef8fcb5411 should resolve this